lib/active_admin/view_helpers/display_helper.rb

```diff
@@ -25,13 +25,7 @@ def display_name_method_for(resource)
         @@display_name_methods_cache ||= {}
         @@display_name_methods_cache[resource.class] ||= begin
           methods = active_admin_application.display_name_methods - association_methods_for(resource)
-          method  = methods.detect{ |method| resource.respond_to? method }
-
-          if method != :to_s || resource.method(method).source_location
```

Comment:
Context: `resource.method(method).source_location` is used to check if it is a native or script defined method.